File: order/src/main/java/om/order/svc/OrderServiceImpl.java
package om.order.svc;

import lombok.RequiredArgsConstructor;
import om.order.dao.OrderRepo;
import om.order.dto.OrderReq;
import om.order.entity.Order;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.UUID;
@Transactional
@Service @RequiredArgsConstructor
public class OrderServiceImpl implements IOrderService {
    private final OrderRepo  orderRepo;
    @Override
    public void createOrder(OrderReq orderReq) {
        Order newOrder = Order.builder()
                .id(UUID.randomUUID().toString())
                .orderNumber(orderReq.orderNumber())
                .itemSkuCode(orderReq.itemSkuCode())
                .pricePerItem(orderReq.pricePerItem())
                .quantity(orderReq.quantity())
                .build();
orderRepo.save(newOrder);

    }
}

File: snapshotV2/order/src/main/kotlin/om/order/service/OrderServiceImpl.kt
package om.order.service

import om.order.dto.OrderReq
import om.order.dto.OrderResp
import om.order.entity.Order
import om.order.repo.OrderRepository
import org.springframework.stereotype.Service

@Service
class OrderServiceImpl(val orderRepository: OrderRepository): OrderService {

    override fun getAllOrders(): Iterable<OrderResp> {
        return orderRepository.findAll().map { it.toResp() }
    }

    override fun createOrder(orderReq: OrderReq):OrderResp {
        return orderRepository.save(Order.fromReq(orderReq)).toResp()
    }
}